src/main/java/duke/commands/DukeCommandFind.java

```diff
@@ -0,0 +1,149 @@
+package duke.commands;
+
+import duke.exceptions.DukeExceptionIllegalArgument;
+import duke.parser.Parser;
+import duke.storage.FileLoader;
+import duke.tasks.DateTask;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Find command.
+ *
+ * Filters tasks according to supplied date range and displays list to user.
+ */
+public class DukeCommandFind extends DukeCommand {
+
+    protected Optional<LocalDateTime> from;
+    protected Optional<LocalDateTime> to;
+    protected String search;
+
+    public DukeCommandFind(String input) throws DukeExceptionIllegalArgument {
+        // Process any keywords first, assumed stripped
+        search = "";
+        if (!input.startsWith("/from ")&&(!input.startsWith("/to "))) {
```

Comment:
Should the && be surrounded by whitespace? Few more instances in this class with operators

Reply:
I wasn't aware this is a coding standard thing?